web-app/src/lib/db/services/credit.service.ts

```diff
@@ -100,7 +98,7 @@ export async function calculateAgentCreditCost(
  */
 export async function calculateCreditCost(
   amounts: { unit: string; amount: number }[],
-): Promise<number> {
+): Promise<bigint> {
```

Comment:
Can you check if this works across boundaries, as serilization works, as bigint is not supported on some function returns

Reply:
It's used three times in the codebase:

1. In the API for credit requests, where it is converted to a string.
2. In the credit service - no issues
3. In the job service - no issues

As far as I know, the only boundary issue is with JSON.

Reply:
@mrgrauel Do we use this function somewhere on the client side? I think there was a bug with serilization, when I tested it for the demo and using bigint. 

If the whole flow with agents works, then you can resolve and merge